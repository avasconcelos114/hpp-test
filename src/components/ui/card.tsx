import * as React from 'react';

import { cn } from '@/lib/utils';

export function Card({ className, ...props }: React.ComponentProps<'div'>) {
  return (
    <div
      data-slot='card'
      className={cn(
        'text-card-foreground flex flex-col gap-6 rounded-xl bg-white p-[25px]',
        className,
      )}
      {...props}
    />
  );
}
