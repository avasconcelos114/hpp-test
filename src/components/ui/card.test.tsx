import React from 'react';
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Card } from './card';

describe('Card', () => {
  it('should render the card', () => {
    render(<Card>Card contents</Card>);
    expect(screen.getByText('Card contents')).toBeDefined();
    expect(screen.getByTestId('card')).toMatchSnapshot();
  });
});
