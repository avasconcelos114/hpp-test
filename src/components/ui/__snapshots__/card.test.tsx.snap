// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should render the card 1`] = `
<div
  class="text-card-foreground flex flex-col gap-6 rounded-xl bg-white p-[25px]"
  data-slot="card"
  data-testid="card"
>
  Card contents
</div>
`;
